libs/context/context.go

```diff
@@ -158,3 +166,17 @@ func InProgressUpgradeFrom(ctx context.Context, from string) bool {
 	}
 	return from == si.version && si.upgrade
 }
+
+// InProgressUpgradeTo returns whether the data in the contexts tells us that the
+// node is restoring from a snapshot as part of an upgrade to a specific version.
+// This ignores suffixes (preview/rc/dev/commit hash) are ignored, only the semver part is checked.
+func InProgressUpgradeTo(ctx context.Context, to string) bool {
```

Comment:
Can you add some unittests in `context_test.go` just to be sure?

Reply:
No problem, done